app/helpers/application_helper.rb

```diff
@@ -22,4 +22,11 @@ def render_flash_message(flash_value)
     end
   end
 
+  def user_has_membership_application?
+    current_user.try(:membership_applications).try(:any?) && current_user.try(:membership_applications).try(:last).try(:persisted?)
+  end
```

Comment:
1) It is not possible for a user to have non-saved member applications that are returned by `#membership_applications`.  That method relies on object ID's being present in the join table - which will not be present if the associated object is not saved.  So, the portion of this statement beginning with `&&` is not needed.

2) The bigger problem is that this level of DB access and model logic is not recommended for views.  That logic should be executed at the controller level, and instance vars used by the view to drive differential logic at that level.